Formula/synergy-core.rb

```diff
@@ -0,0 +1,351 @@
+  # to synergy-core's special OpenSSL exception. The openvpn-openssl-exception
+  # exception is very similar, but it is not the same, so it would be inappropriate
+  # to use it here.
+  #
+  # Unfortunately, writing an exception like this:
+  #     with: :cannot_represent
+  # causes `brew audit` to raise the following error:
+  #     Formula synergy-core contains invalid or deprecated SPDX license exceptions: [:cannot_represent].
+  # As a result, it appears we cannot mention the exception in the `license` definition.
+  license "GPL-2.0-only"
+
+  head "https://github.com/symless/synergy-core.git", branch: "master"
+
+  depends_on "cmake" => :build
+  depends_on "openssl@3"
+  depends_on "pugixml"
+  depends_on "qt@5"
+
+  on_linux do
+    depends_on "pkgconfig" => :build
+    depends_on "gdk-pixbuf"
+    depends_on "glib"
+    depends_on "libnotify"
+    depends_on "libxkbfile"
+  end
+
+  # Build against our pugixml rather than the version included in the
+  # synergy-core repository. This is particularly necessary because the
+  # build process for synergy-core attempts to download the source code of
+  # pugixml and build it. This is not consistent with how things are done
+  # in brew, but it appears to be an intentional decision by upstream.
+  #
+  # Also, synergy-core's attempt to download the source code of pugixml
+  # works only if a `.git` directory exists in the source tree. Unless we're
+  # building with a git clone (rather than a release tarball), this doesn't
+  # even work correctly.
+  #
+  # The changelog for v1.13.0-stable includes the following item:
+  #   "[A]void using brew for build"
+  # As a result, it appears that this patch will not be accepted upstream.
+  #
+  # However, we still need to make this change to build this program, so
+  # this patch is unavoidable for now.
+  patch :DATA
+
+  def install
+    system "cmake", "-S", ".", "-B", "build", *std_cmake_args, "-DBUILD_TESTS:BOOL=OFF"
+
+    # Use the standard brew installation path.
+    inreplace "CMakeLists.txt",
+              "set (SYNERGY_BUNDLE_DIR ${CMAKE_BINARY_DIR}/bundle)",
+              "set (SYNERGY_BUNDLE_DIR ${CMAKE_INSTALL_PREFIX}/bundle)"
+
+    # This project has separate make and install steps.
+    system "make", "-C", "build"
+    system "make", "install/strip", "-C", "build"
+
+    # On macOS, create symlinks to launch the installed binaries.
+    # This is not necessary on Linux. On Linux, the `make install/strip`
+    # command already places these binaries directly under `prefix/bin`.
+    if OS.mac?
+      bin.install_symlink prefix/"bundle/Synergy.app/Contents/MacOS/synergy"  # main GUI program
+      bin.install_symlink prefix/"bundle/Synergy.app/Contents/MacOS/synergys" # server
+      bin.install_symlink prefix/"bundle/Synergy.app/Contents/MacOS/synergyc" # client
+    end
+  end
+
+  service do
+    run [opt_bin/"synergy"]
+    run_type :immediate
+  end
+
+  def caveats
```

Comment:
Nothing here seems special for Homebrew, so this should all be in the upstream docs instead of a caveat.

Reply:
I've deleted the Linux caveat because it is nothing specific to brew, as you say.

For the macOS caveats, I've added an explanation in the comments of why the information is specific to brew:

```ruby
    # The official Synergy binaries built by Symless (for which it charges money)
    # are signed by a trusted code-signing certificate. For that reason, the user
    # only has to add Synergy.app to the 'Accessibility' permissions list once,
    # and it will work forever, even after upgrading or overwriting the app (so
    # long as the new versions are also signed by the same certificate).
    #
    # Also, with the official binaries, the user only ever has to add one entry
    # to the 'Accessibility' permissions list (namely, /Applications/Synergy.app),
    # whereas with our version, it is sometimes necessary to add more than one
    # program (or app bundle) to the permissions list (as explained below).
    #
    # This `caveats` block explains all of the side effects of our macOS binaries
    # not being signed by a trusted certificate. This information is specific to
    # the brew installation and is not included in upstream docs because upstream
    # avoids many of these inconveniences by shipping binaries signed with a
    # trusted certificate (and charging money for those binaries).
```